notifications/views.py

```diff
@@ -163,3 +174,51 @@ def live_unread_notification_list(request):
         'unread_list': unread_list
     }
     return JsonResponse(data)
+
+
+def live_notification_list(request):
+    if not request.user.is_authenticated():
+        data = {
+           'all_count':0,
+           'all_list':[]
+        }
+        return JsonResponse(data)
+
+    try:
+        num_to_fetch = request.GET.get('max', 5)  # If they don't specify, make it 5.
+        num_to_fetch = int(num_to_fetch)
+        num_to_fetch = max(1, num_to_fetch)  # if num_to_fetch is negative, force at least one fetched notifications
+        num_to_fetch = min(num_to_fetch, 100)  # put a sane ceiling on the number retrievable
+    except ValueError:
+        num_to_fetch = 5  # If casting to an int fails, just make it 5.
+
+    before_datetime = request.GET.get('before')
+
+    all_list = []
+
+    all_notifications_qa = request.user.notifications.all()
+
+    if before_datetime:
+        all_notifications_qa = all_notifications_qa.filter(timestamp__lt=dateutil.parser.parse(before_datetime))
+
+
+    for n in all_notifications_qa[0:num_to_fetch]:
+        struct = model_to_dict(n)
+        if n.actor:
+            struct['actor'] = str(n.actor)
```

Comment:
these should be unicode casts. but also, why cast at all. these should already be strings...